proposals/p2240.md

```diff
@@ -256,63 +296,71 @@ expansion with arity 0 is rewritten to `false` (the identities of the respective
 logical operations). A `...,` expansion with arity 0 is rewritten to `,`, and
 any `,` immediately before or after the expansion is removed.
 
-Even though an `[:]` operator can only appear within a pack expansion, it is
-possible to apply these rewrite semantics to an isolated subexpression that
-contains a `[:]` operator but not the enclosing pack operator. In those cases,
-the result of the rewrite is an expression headed by a `[:]` operator, which
-means that its value is a pack. Thus, we can say that
-`([:]("a", "b", "c"), [:](1, 2, 3))` rewrites to
-`[:](("a", 1), ("b", 2), ("c", 3))`.
-
-### Symbolic expression semantics and pack values
-
-Under some circumstances, a variadic expression must be evaluated symbolically,
-because the values of one or more of its expansion arguments is not yet known.
-For example, this is often needed when variadic operations are used in a type
-position. To support that, we need to introduce some new kinds of values to
-represent the inputs and outputs of indefinite variadic operations.
-
-A _pack element_ represents a sequence of an indefinite number of values that
-all match a single pattern. The pattern and the size of the sequence are
-represented as
+### Reified semantics and pack values
+
+Unlike syntactic semantics, the reified semantic model is used at compile time,
+when values may be unknown, or contain unknowns. Crucially, that can include
+pack arities, so our representation of packs must be able to accommodate that.
+
+A _pack element_ represents a sequence of some definite or indefinite number of
+values that all match a single value, which is called the _kernel_. During
```

Comment:
I'm having trouble with this sentence:

- Is it framed like the definition of a pack element, but instead seems to tell you what they do.
- Part of the confusion here seems to be that this is explicitly in the context where the elements of a pack are unknown?
- This is talking about values matching values? I thought patterns match values.
- In general, some context seems to be missing.
- Unclear what the difference between a pack element and a kernel is here.

Maybe it could be explained in the context of an example?